### PR TITLE
97199 cst add reference id to evidence submissions table when uploading document

### PR DESCRIPTION
**Pull request by samcoforma**
## Summary

- Updated `document_upload.rb` perform method to populate the `request_id` when there is a success response
- Updated `document_upload.rb` to have a method called `record_evidence_submission` 
- Added migration to add `request_id` to the `evidence_submissions` table
- Updated the benefits documents service and removed `record_evidence_submission` to `document_upload.rb` so that when submit_document calls the perform method in `document_upload.rb` the `record_evidence_submission`  method is called
- Created a constants file for the upload statuses
- Updated benefits documents service test to check the perform method returns a string
- Updated the document upload test to test that that a file is uploaded successfully to lighthouse and a request id is returned
- Updated the document upload test to test that an error is raised when lighthouse returns a failure response 

## Related issue(s)

- *Link to ticket created in va.gov-team https://github.com/department-of-veterans-affairs/va.gov-team/issues/97199

## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: lib/lighthouse/benefits_documents/service.rb [hunk 1]
```diff
@@ -75,7 +75,7 @@ def submit_document(file, file_params, lighthouse_client_id = nil)
       if Flipper.enabled?(:cst_synchronous_evidence_uploads, @user)
         Lighthouse::DocumentUploadSynchronous.upload(user_icn, document_data.to_serializable_hash)
       else
-        Lighthouse::DocumentUpload.perform_async(user_icn, document_data.to_serializable_hash)
+        Lighthouse::DocumentUpload.perform_in(1.minute, user_icn, document_data.to_serializable_hash)
```
Comment (jerekshoe) [inline lib/lighthouse/benefits_documents/service.rb:78]: What's the reason for this change?

Comment (samcoforma) [reply lib/lighthouse/benefits_documents/service.rb]: @jerekshoe We reverted to using `perform_async` by moving the `EvidenceSubmission` record creation responsibility into `DocumentUpload` and out of `Service`

File: lib/lighthouse/benefits_documents/service.rb [hunk 2]
```diff
@@ -64,24 +49,38 @@ def cleanup_after_upload
...
       user_icn = @user.icn
+      user_account_uuid = @user.user_account_uuid
       document_data = build_lh_doc(file, file_params)
+      claim_id = file_params[:claimId] || file_params[:claim_id]
+      tracked_item_id = file_params[:trackedItemIds] || file_params[:tracked_item_ids]
+
+      unless claim_id
+        raise Common::Exceptions::InternalServerError,
+              ArgumentError.new("Claim with id #{claim_id} not found")
+      end
```
Comment (shaunburdick) [inline lib/lighthouse/benefits_documents/service.rb:60]: Wouldn't `#{claim_id}` be empty here so adding it to the log message would just be a empty string?

Comment (samcoforma) [reply lib/lighthouse/benefits_documents/service.rb:60]: Yup, good catch. I'll fix that. Also, correct about it being inherited code 😆

Comment (samcoforma) [reply lib/lighthouse/benefits_documents/service.rb]: @shaunburdick This error has been changed to say that a claim id is required without printing a claim id in the error

**Review by shaunburdick — APPROVED** [review]
LGTM
